### PR TITLE
Resolve #6807, remove all OSVDB references.

### LINKED ISSUE #6807: Remove all OSVDB references
Since [OSVDB is no longer around](https://blog.osvdb.org/2016/04/05/osvdb-fin/), all the references are now bad links, and we need to remove them.

### PR DESCRIPTION
**Pull request by wchen-r7**
## What this patch does

This removes all the OSVDB references, because OSVDB is no longer a vulnerability database. Please also see: https://blog.osvdb.org/2016/04/05/osvdb-fin/

Resolve #6807
## Verification steps
- [x] Travis must be green
- [x] Start msfconsole
- [x] You should not see any errors

If you would like to manually read the changes, since github cannot show you all the changes files, you probably wanna do: `git diff > /tmp/osvdb.patch`, and then look at that file.

### CODE REVIEW

File: modules/auxiliary/admin/edirectory/edirectory_dhost_cookie.rb [hunk 1]
```diff
@@ -22,7 +22,6 @@ def initialize(info = {})
       },
       'References'     =>
```
Comment (FireFart) [inline modules/auxiliary/admin/edirectory/edirectory_dhost_cookie.rb]: this one for example has no references after deleting the osvdb entry. Should we remove the whole References hash or is at least one reference required?

Comment (wchen-r7) [inline modules/auxiliary/admin/edirectory/edirectory_dhost_cookie.rb]: Good eye. It seems to load fine, so I didn't try to change it. Let me know how you want it :-)

Comment (FireFart) [inline modules/auxiliary/admin/edirectory/edirectory_dhost_cookie.rb]: As there are many modules without references, i would simply remove the references hash when there are no references left.

Comment (wchen-r7) [inline modules/auxiliary/admin/edirectory/edirectory_dhost_cookie.rb]: ok cool

**Comment by FireFart** [comment]
@wchen-r7 spotted a few other empty references:
```
/modules/auxiliary/admin/edirectory/edirectory_dhost_cookie.rb
/modules/auxiliary/dos/windows/smb/rras_vls_null_deref.rb
/modules/auxiliary/dos/windows/smb/vista_negotiate_stop.rb
/modules/auxiliary/scanner/http/http_put.rb
/modules/auxiliary/scanner/http/manageengine_deviceexpert_traversal.rb
/modules/exploits/apple_ios/ssh/cydia_default_ssh.rb
/modules/exploits/linux/http/hp_system_management.rb
/modules/exploits/linux/http/wanem_exec.rb
/modules/exploits/linux/local/hp_smhstart.rb
/modules/exploits/multi/browser/opera_configoverwrite.rb
/modules/exploits/windows/browser/athocgov_completeinstallation.rb
/modules/exploits/windows/ftp/sasser_ftpd_port.rb
/modules/exploits/windows/imap/novell_netmail_auth.rb
```

**Comment by wchen-r7** [comment]
Ahh, ok. I'll fix them. Thanks!

**Comment by wchen-r7** [comment]
ok done!

**Comment by FireFart** [comment]
@wchen-r7 there are also some references to OSVDB in `lib/msf/core/db_manager/import/nikto.rb`. Should these be also taken into account?

Otherwise this change looks good to me. Is it ok to merge or is there some other work to do like adopting pro?

**Comment by wchen-r7** [comment]
@FireFart Good question, I didn't touch nikto.rb because I wasn't sure what to do with that. But since you asked, I guess we should :-)

**Comment by wchen-r7** [comment]
Ok that is gone.

**Comment by wchen-r7** [comment]
>  Is it ok to merge or is there some other work to do like adopting pro?

Yes, there's some stuff to do on Pro, mostly tests. Let me ask the team on Monday and we'll see what to do from there. Thanks!

**Comment by FireFart** [comment]
awesome! so let's wait for the pro guys :)

**Comment by bcook-r7** [comment]
Hey @wvu-r7 , could you check out the pro impact here + cucumber adjustments?

**Comment by wvu-r7** [comment]
Will do.

**Comment by bcook-r7** [comment]
Let's target deleting this in 2 weeks. We may change course if there is an odd issue (or if a mirror / champion arrives)

**Comment by wchen-r7** [comment]
What's the progress for this?

**Comment by bcook-r7** [comment]
Landing this to make progress.

**Comment by egypt** [comment]
@bcook-r7 why was this reverted?

**Comment by bcook-r7** [comment]
It broke links between metasploit and nexpose where OSVDB reference was the only link. Also, we found the all of the OSVDB information. Plus, why delete metadata - it doesn't actually hurt anything.

**Pull request merged**